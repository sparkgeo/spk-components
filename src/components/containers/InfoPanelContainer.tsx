import { ReactElement } from "react";
import styles from "./InfoPanelContainer.module.css";

type InfoPanelProps = {
    children: ReactElement,
    position: string
}

export const InfoPanelContainer = ({children, position='centerRight'}:InfoPanelProps) => (
    <div className={`${styles.infoPanelContainer} ${styles[position]}`}>
        <div>{children}</div>
    </div>
);
