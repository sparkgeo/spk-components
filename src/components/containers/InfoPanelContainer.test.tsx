import { render } from "@testing-library/react";
import { InfoPanelContainer } from "./InfoPanelContainer";

describe("InfoPanelContainer", () => {
    it("renders with correct position class and children", () => {
        const StubChildComponent = () => (<div>This is a stubbed child element</div>)

        const {container} = render(
            <InfoPanelContainer data-testid='infoPanel' position="topLeft">
                <StubChildComponent/>
            </InfoPanelContainer>
        );

        expect(container).toMatchSnapshot();
    });
});