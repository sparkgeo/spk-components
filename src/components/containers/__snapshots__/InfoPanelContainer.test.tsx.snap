// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoPanelContainer > renders with correct position class and children 1`] = `
<div>
  <div
    class="_infoPanelContainer_0bef1b _topLeft_0bef1b"
    data-testid="infoPanel"
  >
    <div>
      <div>
        This is a stubbed child element
      </div>
    </div>
  </div>
</div>
`;
